src/modules/records/publisher/publisher.js

```diff
@@ -23,6 +23,80 @@ export default class Publisher extends OIPRecord {
         facebook: undefined
       }
     }
+    this.signature = undefined
+  }
+
+  /**
+   * Create message to use for Publisher signature
+   * @see `alias-address-timestamp`: https://oip.wiki/Message_protocol#Publisher_Register_signature
+   * @return {string}
+   */
+  createPreimage () {
+    if (!this.getTimestamp()) { this.setTimestamp(Date.now()) }
+
+    let preimage = `${this.getAlias() || ''}-${this.getMainAddress()}-${this.getTimestamp()}`
+    this.preimage = preimage
+    return preimage
+  }
+
+  /**
+   * Check if the Publisher is Valid and has all the required fields to be registered
+   * @return {StatusObject}
+   */
+  isValid () {
+    if (!this._publisher.timestamp || isNaN(this._publisher.timestamp)) {
+      return { success: false, error: 'Timestamp is a Required Field!' }
+    }
+    if (!this._publisher.floAddress || this._publisher.floAddress === '') {
+      return { success: false, error: 'floAddress is a Required Field! Please set it using `setMainAddress`!' }
+    }
+
+    return { success: true }
+  }
+
+  toJSON() {
+    return JSON.parse(JSON.stringify({ pub: this._publisher }))
+  }
+
+  serialize (method) {
+    // convert this to json and extract artifact
+    let publisherJSON = this.toJSON()
+    let pub = publisherJSON.pub
+
+    /* @ToDo: Reimplement publishing of `authorized`, `info`, and `verificatiton` once they are added to be supported by OIP daemon */
+    // Remove `pub.authorized` section
+    pub.authorized = undefined
+    // Remove `pub.info` section
+    pub.info = undefined
+    // Remove `pub.verification` section
+    pub.verification = undefined
+
+    // setup initial object
+    let pubMessage = { oip042: {} }
```

Comment:
should we make the `oip042` prefix be a variable so its dynamic for future versions?

Reply:
Yeah we should make it dynamic in the future, but for now I am okay with it only supporting `oip042`.